detection_rules/rule.py

```diff
@@ -412,6 +412,9 @@ def validator(self) -> Optional[QueryValidator]:
     def validate_query(self, meta: RuleMeta) -> None:
         validator = self.validator
         if validator is not None:
+            # TODO: Decide on whether to only validate_integrations or also validate
+            if meta.integration:
+                return validator.validate_integration(self, meta)
```

Comment:
Design Q: Should we do one or the other, OR both?

Reply:
the `validator.validate` method should know that if it parses an integration index from the `index` array, then it should pass that datasource validation to `validate_integration`. In other words, it should be either or, but exhaustive for all datasources